### ISSUE TITLE
CONDA_GSL issues

### ISSUE THREAD

**Issue by jeromekelleher** [bioconda/bioconda-utils#36]
There seems to be an issue with GSL in bioconda at the moment. My package [msprime](https://github.com/bioconda/bioconda-recipes/blob/master/recipes/msprime/meta.yaml#L23) depends on GSL and uses the CONDA_GSL variable. This worked well previously. Now though, if I delete the gsl package and run ``conda install msprime`` I get 
```
$ conda install msprime
Fetching package metadata .............
Solving package specifications: .
Error: Dependency missing in current osx-64 channels: 
  - msprime -> gsl 1.16*

You can search for packages on anaconda.org with

    anaconda search -t conda gsl

You may need to install the anaconda-client command line client with

    conda install anaconda-client
```
This is on OSX.

The problem is perhaps to do with the fact the conda-forge is now using gsl version 2.1:

https://anaconda.org/conda-forge/gsl

I guess I have two questions:

1. Why do we use CONDA_GSL and pin the version rather than just having an ordinary dependency on gsl (I just followed the pattern I found in other recipes here, monkey-see-monkey-do)?
2. Should we update CONDA_GSL to 2.1 and trigger rebuilds of all the packages that use it?

Thanks to @molpopgen for pointing this problem out to me.

**Comment by johanneskoester** [comment]
This is annoying. There was a gsl 1.16 in the default channel when we pinned it. Apparently, it has been deleted. Of course, this is not how Continuum should manage the channel, because it defeats reproducibility in science to not being able to rely on old versions.

Regarding your questions:
1. the pinning is necessary because you usually link against the library and new versions can be binary incompatible. This way of pinning allows us to update all GSL dependent packages in one pass. It also pays of in this case, because I can just change the variable definition in order to fix all dependent packages (see PR #3041).
2. see PR #3041

**Comment by jeromekelleher** [comment]
Thanks @johanneskoester, this is super helpful!
